src/main/java/core/basesyntax/model/Operation.java

```
@@ -0,0 +1,29 @@
+package core.basesyntax.model;
+
+import java.util.Arrays;
+
+public enum Operation {
+
+    BALANCE("b"),
+    SUPPLY("s"),
+    PURCHASE("p"),
+    RETURN("r");
+
+    private String type;
+
+    Operation(String type) {
+        this.type = type;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public static Operation getOperationType(String type) {
+        return Arrays.stream(Operation.values())
+                .filter(operation -> operation.getType().equalsIgnoreCase(type))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("error : element not found."));

```

This violates checklist item: "Operation code must be one of b, s, p, r; otherwise throw informative RuntimeException." Currently getOperationType(...) uses .orElseThrow(() -> new IllegalArgumentException("error : element not found.")) which produces a generic message that doesn't include the invalid value or helpful context. Please throw an informative exception (for example: `throw new IllegalArgumentException("Unknown operation type: " + type);`).